src/Components/Server/src/DependencyInjection/RazorComponentsServiceCollectionExtensions.cs

```diff
@@ -101,7 +101,10 @@ private static void AddStandardRazorComponentsServices(IServiceCollection servic
             services.TryAddScoped(s => s.GetRequiredService<ICircuitAccessor>().Circuit);
             services.TryAddScoped<IJSRuntimeAccessor, DefaultJSRuntimeAccessor>();
             services.TryAddScoped(s => s.GetRequiredService<IJSRuntimeAccessor>().JSRuntime);
-            services.TryAddScoped<IUriHelper, RemoteUriHelper>();
+            services.TryAddScoped<UriHelperAccessor>();
+            services.TryAddScoped<IUriHelper>(s => s.GetRequiredService<UriHelperAccessor>().UriHelper);
```

Comment:
One dangerous thing about this pattern is that it makes it seem like you can **change** `UriHelper`. but you SUPER CANT. Components that get `IUriHelper` from services will only see the one that's available when the initialize.

If you wanted to harden this you could do the following:
1. Expose through services an `IUriHelper` that delegates to `UriHelperAccessor`
2. Now you can update `UriHelperAccessor.UriHelper` as needed, and consumers instantly see the updates.

Reply:
I think it's just nicer in general when we think about separate an API that is **for user code** from how we're going to implement it. Instead of creating 3 different URI helpers, maybe it's just a facade (etc)